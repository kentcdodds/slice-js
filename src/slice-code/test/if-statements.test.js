import {runAllCombosTests, snapSlice} from './helpers/utils'

// TODO This is broken
xit(
  'test case',
  snapSlice(require.resolve('./fixtures/if-statements'), ({ifElseIfElse}) => {
    ifElseIfElse(true, true)
    ifElseIfElse(true, false)
    ifElseIfElse(false, false)
  })
)

runAllCombosTests({
  filename: require.resolve('./fixtures/if-statements'),
  methods: [
    {methodName: 'ifOnly', possibleArguments: [[true], [false]]},
    {methodName: 'ifElse', possibleArguments: [[true], [false]]},
    {
      methodName: 'ifElseIfElse',
      possibleArguments: [
        [true, true], [true, false], [false, true], [false, false],
      ],
    },
    {
      methodName: 'nestedIf',
      possibleArguments: [
        [true, true], [true, false], [false, true], [false, false],
      ],
    },
  ],
})
