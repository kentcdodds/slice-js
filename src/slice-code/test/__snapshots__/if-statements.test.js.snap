exports[`test ifElse(false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElse(passIf) {
  return !passIf;
}"
`;

exports[`test ifElse(true) && ifElse(false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElse(passIf) {
  if (passIf) {
    return passIf;
  } else {
    return !passIf;
  }
}"
`;

exports[`test ifElse(true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElse(passIf) {
  return passIf;
}"
`;

exports[`test ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  return !passIf1 && !passIf2;
}"
`;

exports[`test ifElseIfElse(false, true) && ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf2) {
    return passIf2;
  } else {
    return !passIf1 && !passIf2;
  }
}"
`;

exports[`test ifElseIfElse(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  return passIf2;
}"
`;

exports[`test ifElseIfElse(true, false) && ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else {}
}"
`;

exports[`test ifElseIfElse(true, false) && ifElseIfElse(false, true) && ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else if (passIf2) {
    return passIf2;
  } else {
    return !passIf1 && !passIf2;
  }
}"
`;

exports[`test ifElseIfElse(true, false) && ifElseIfElse(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else {
    return passIf2;
  }
}"
`;

exports[`test ifElseIfElse(true, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  return passIf1;
}"
`;

exports[`test ifElseIfElse(true, true) && ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else {}
}"
`;

exports[`test ifElseIfElse(true, true) && ifElseIfElse(false, true) && ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else if (passIf2) {
    return passIf2;
  } else {
    return !passIf1 && !passIf2;
  }
}"
`;

exports[`test ifElseIfElse(true, true) && ifElseIfElse(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else {
    return passIf2;
  }
}"
`;

exports[`test ifElseIfElse(true, true) && ifElseIfElse(true, false) && ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else {}
}"
`;

exports[`test ifElseIfElse(true, true) && ifElseIfElse(true, false) && ifElseIfElse(false, true) && ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else if (passIf2) {
    return passIf2;
  } else {
    return !passIf1 && !passIf2;
  }
}"
`;

exports[`test ifElseIfElse(true, true) && ifElseIfElse(true, false) && ifElseIfElse(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  } else {
    return passIf2;
  }
}"
`;

exports[`test ifElseIfElse(true, true) && ifElseIfElse(true, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  return passIf1;
}"
`;

exports[`test ifElseIfElse(true, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifElseIfElse(passIf1, passIf2) {
  return passIf1;
}"
`;

exports[`test ifOnly(false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifOnly(passIf) {}"
`;

exports[`test ifOnly(true) && ifOnly(false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifOnly(passIf) {
  if (passIf) {
    return passIf;
  }
}"
`;

exports[`test ifOnly(true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function ifOnly(passIf) {
  if (passIf) {
    return passIf;
  }
}"
`;

exports[`test nestedIf(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {}"
`;

exports[`test nestedIf(false, true) && nestedIf(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {}"
`;

exports[`test nestedIf(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {}"
`;

exports[`test nestedIf(true, false) && nestedIf(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, false) && nestedIf(false, true) && nestedIf(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, false) && nestedIf(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, true) && nestedIf(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    if (passIf2) {
      return passIf2;
    }
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, true) && nestedIf(false, true) && nestedIf(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    if (passIf2) {
      return passIf2;
    }
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, true) && nestedIf(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    if (passIf2) {
      return passIf2;
    }
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, true) && nestedIf(true, false) && nestedIf(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    if (passIf2) {
      return passIf2;
    }
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, true) && nestedIf(true, false) && nestedIf(false, true) && nestedIf(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    if (passIf2) {
      return passIf2;
    }
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, true) && nestedIf(true, false) && nestedIf(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    if (passIf2) {
      return passIf2;
    }
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, true) && nestedIf(true, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    if (passIf2) {
      return passIf2;
    }
    return passIf1;
  }
}"
`;

exports[`test nestedIf(true, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse, nestedIf };

function nestedIf(passIf1, passIf2) {
  if (passIf1) {
    if (passIf2) {
      return passIf2;
    }
    return passIf1;
  }
}"
`;
